### DISCUSSION TITLE
v1.7.0

### DISCUSSION THREAD

**Discussion by cube2222** [opentofu/opentofu#1588] [Announcements]
We're proud to announce that OpenTofu 1.7.0 is now officially out! 🎉

**What’s New?**
- State Encryption - Protect your precious state files with end-to-end encryption.
- Dynamic Provider-defined Functions - Author custom functions as part of providers.
- Declarative removed Blocks - No more fighting with the CLI, execute your removals declaratively!
- Loopable Import Blocks - Streamline importing multiple resources.
- ...and much, much more!

See the launch post on our blog: https://opentofu.org/blog/opentofu-1-7-0/

For all the features, see the [detailed changelog](https://github.com/opentofu/opentofu/blob/ba4c3a5b462129cd61644a3ae760d746dc12b8fc/CHANGELOG.md).

You can find the full diff [here](https://github.com/opentofu/opentofu/compare/v1.6...v1.7.0?expand=1).

<hr /><em>This discussion was created from the release <a href='https://github.com/opentofu/opentofu/releases/tag/v1.7.0'>v1.7.0</a>.</em>

**Comment by adv4000** [comment]
Time to record tutorial about migration to OpenTofu :)
